src/soilbiogeochem/SoilBiogeochemDecompCascadeCNMod.F90

```diff
@@ -933,13 +998,21 @@ subroutine decomp_rate_constants_cn(bounds, &
                 do fc = 1,num_soilc
                    c = filter_soilc(fc)
                    decomp_k(c,j,i_cwd) = k_frag * t_scalar(c,j) * w_scalar(c,j) * depth_scalar(c,j) * o_scalar(c,j) / dt
+                  if(use_soil_matrixcn)then
```

Comment:
The "if(use_soil_matrixcn)" if statement should go around the entire if statement

Reply:
Oh, wait decomp_k is being calculated with use_soil_matrixcn on or off, didn't see that. Sorry...

So I suppose it should stay as it is. This does mean the if statement is calculated inside the loop, but because of predictive branching it should still be fast.